app/controllers/login.rb

```diff
@@ -0,0 +1,23 @@
+class MakersBnb < Sinatra::Base
+
+  get '/' do
+    redirect '/spaces'
+  end
+
+  get '/sessions/new' do
+    erb :'sessions/new'
+  end
+
+  post '/sessions' do
+    @user = User.authenticate(params[:email], params[:password])
+    if @user
+      session[:user_id] = @user.id
+      flash[:notice] = flash[:notice] = "Welcome, #{@user.name}"
+      redirect '/spaces'
+    else
+      flash.now[:errors] = ['The email or password is incorrect']
+      erb :'sessions/new'
+    end
+  end
+
+end
```

Comment:
Beautiful controller. Brings a tear to my eye.

I guess you could do this in `post '/sessions'`:

```ruby
if @user = User.authenticate(params[:email], params[:password])
  #etc
```